### ISSUE TITLE
Problems when using proxy server

### ISSUE THREAD

**Issue by mdpi-symfony** [Gregwar/CaptchaBundle#178]
Hello all,
we are experiencing some strange behavior from our captcha checks (we are not really sure whether this could related to the captcha or the captcha bundle - or whether this could be something related to our server settings in general).

The problem is simply that the captcha is somewhat randomly failing the checks when entering correct string.

Our setup is roughly like this:
- we have HAproxy server that redirects the requests to 5 servers (i.e. initial form loading and captcha value storage can happen in server 1 and captcha validation in server 2).
- we have single captcha on the page when error happens (the captcha form itself is loaded via ajax).
- testing the captcha separately connecting to each individual server (via hosts file settings) seems to always work

We have logged how the requests are distributed to the servers in order to trying to identify whether the errors happen always with some specific server and drew a blank there - there didn't seem to be any logic that we could distinguish which cases were succeeding and which failing (we did catch couple of cases where the sequence was same and in these cases the result was also always the same).

any help / insight / idea would be much appreciated.

br,
mdpi devs

**Comment by Gregwar** [comment]
Hello

How does your session system work?
Since the phrase is stored in the session when it is generated to be checked later, the server that validate the phrase should be able to access the session value generated by the one that issued the captcha

**Comment by mdpi-symfony** [comment]
Thank you for your prompt reply Gregwar.

We are using a separate server which storres all sessions, so it is centralized.

**Comment by Gregwar** [comment]
Is there delays or cache in this centralization system?

From the client point of view, the server hostname is always the same right?

Can you have a look in the session between the two requests to check that everything is ok? (You should see the correct phrase)

You could also do another more minimalistic test without ajax loading (and without as_url also for instance)

**Comment by mdpi-symfony** [comment]
Thank you for your prompt reply Gregwar.

We are using a separate server which storres all sessions, so it is centralized.

**Comment by mdpi-symfony** [comment]
Forgot to answer the rest, sorry!

We use memcache for centralized session storage.

The server hostname is always the same.

Some help on how to have a look in the sessions between the two requests?

Will try the last point with no ajax use.

**Comment by Gregwar** [comment]
Can you try disabling temporary memcache to see if it changes anything?

To inspect session I gues you can add a (temporary secret debug) controller that dumps session values

**Comment by mdpi-symfony** [comment]
ok, so latest updates. It looks very likely that the problem was caused by something in our server configuration/setup, i.e. we did some shuffling with the used servers and those seemed to fix this issue.

The biggest obvious difference in our new server setup is that we are now using same ubuntu versions on all of the servers (16.04) whereas previously we had also couple servers with older versions (14.04) in the pool - and according to the initial (albeit quite limited) testings, at least one of the servers with older ubuntu version seemed to be always participating in the failing cases.

This issue is now resolved from our side and can be closed. Thank you very much for your prompt answers and support - it is much appreciated.
